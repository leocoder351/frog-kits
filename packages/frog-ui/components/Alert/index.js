import React from 'react';
import t from 'prop-types';
import classNames from 'classnames';

const Alert = props => {
  const { type, message } = props;
  return (
    <div 
      className={classNames('j-alert', {
        [`j-alert-${type}`]: true
      })}>
      {message}
    </div>
  )
}

Alert.propTypes = {
  type: t.oneOf(['success', 'info', 'warning', 'error']),
  message: t.string
};

Alert.defaultProps = {
  type: 'success'
};

export default Alert;