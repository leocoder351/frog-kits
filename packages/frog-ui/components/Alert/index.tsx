import React from 'react';
import classNames from 'classnames';

type Props = {
  type?: 'success' | 'info' | 'warning' | 'error';
  message: string;
};

const Alert: React.FC<Props> = ({ type = 'default', message }) => {
  return (
    <div
      className={classNames('frog-alert', {
        [`frog-alert-${type}`]: true,
      })}
    >
      {message}
    </div>
  );
};

export default Alert;
