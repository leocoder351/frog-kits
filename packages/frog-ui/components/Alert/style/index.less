@import '../../style/index.less';
@import "../../style/color/default";

.frog-alert {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  margin-bottom: 16px;
  padding: 8px 15px;
  color: rgba(0,0,0,.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  word-wrap: break-word;
  list-style: none;
  border: 1px solid;
  border-radius: 2px;
  font-feature-settings: "tnum";

  &-success {
    background: @cyan2;
    border-color: @cyan4;
  }

  &-info {
    background: @blue2;
    border-color: @blue4;
  }

  &-warning {
    background: @yellow2;
    border-color: @yellow4;
  }

  &-error {
    background: @red2;
    border-color: @red4;
  }
}