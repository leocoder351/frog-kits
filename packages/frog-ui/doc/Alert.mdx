---
name: Alert 警告提示
route: /alert
menu: 反馈
---

import { Playground, Props } from 'docz'
import { Alert } from '../components/';
import '../components/Alert/style';

# Alert
警告提示，展现需要关注的信息。

<Props of={Alert} />

## 基本用法

<Playground>
  <Alert message="Success Text" type="success" />
  <Alert message="Info Text" type="info" />
  <Alert message="Warning Text" type="warning" />
  <Alert message="Error Text" type="error" />
</Playground>