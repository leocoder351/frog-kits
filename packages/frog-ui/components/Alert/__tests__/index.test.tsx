import React from 'react';
import renderer from 'react-test-renderer';
import Alert from '../index';

describe('Component <Alert /> Test', () => {
  test('should render default', () => {
    const component = renderer.create(<Alert message="default" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render specific type', () => {
    const types: any[] = ['success', 'info', 'warning', 'error'];
    const component = renderer.create(
      <>
        {types.map((type) => (
          <Alert key={type} type={type} message={type} />
        ))}
      </>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
