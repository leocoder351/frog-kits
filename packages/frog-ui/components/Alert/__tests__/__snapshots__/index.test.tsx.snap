// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Alert /> Test should render default 1`] = `
<div
  className="frog-alert frog-alert-default"
>
  default
</div>
`;

exports[`Component <Alert /> Test should render specific type 1`] = `
Array [
  <div
    className="frog-alert frog-alert-success"
  >
    success
  </div>,
  <div
    className="frog-alert frog-alert-info"
  >
    info
  </div>,
  <div
    className="frog-alert frog-alert-warning"
  >
    warning
  </div>,
  <div
    className="frog-alert frog-alert-error"
  >
    error
  </div>,
]
`;
